From 2024562ed45d905bee00aea923c218c493c09d27 Mon Sep 17 00:00:00 2001
From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Fri, 22 Jul 2016 13:10:02 -0400
Subject: [PATCH] rtl8xxxu: Add simple rtl8188eu_rf_on() routine

It is not obvious from the vendor driver if we need more than this.

Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
---
 drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_8188e.c | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_8188e.c
+++ b/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_8188e.c
@@ -1096,6 +1096,11 @@ exit:
 	return ret;
 }
 
+static void rtl8188e_enable_rf(struct rtl8xxxu_priv *priv)
+{
+	rtl8xxxu_write8(priv, REG_TXPAUSE, 0x00);
+}
+
 static void rtl8188e_usb_quirks(struct rtl8xxxu_priv *priv)
 {
 	u16 val16;
@@ -1127,6 +1132,7 @@ struct rtl8xxxu_fileops rtl8188eu_fops =
 	.phy_iq_calibrate = rtl8188eu_phy_iq_calibrate,
 	.config_channel = rtl8188eu_config_channel,
 	.parse_rx_desc = rtl8xxxu_parse_rxdesc16,
+	.enable_rf = rtl8188e_enable_rf,
 	.usb_quirks = rtl8188e_usb_quirks,
 	.update_rate_mask = rtl8xxxu_gen2_update_rate_mask,
 	.report_connect = rtl8xxxu_gen2_report_connect,
